Techion.rte/Devices/Weapons/NanoRifle/NanoShot.lua

```diff
@@ -19,22 +19,37 @@ function Create(self)
 			end
 		end
 	end
-	
 	--Speed at which this can actually activate.
 	self.speedThreshold = 100;
 
 	self.deleteNextFrame = false;
 	--Check backward.
 	self.CheckCollision(Vector(self.Vel.X, self.Vel.Y):RadRotate(math.pi));
+
+	self.trailLength = 50;
+
+	self.trailPar = CreateMOPixel("Nano Rifle Trail Glow");
+	self.trailPar.Pos = self.Pos;
+	self.trailPar.Vel = self.Vel * 0.1;
+	self.trailPar.Lifetime = 60;
+	MovableMan:AddParticle(self.trailPar);
+
+	self.lastVel = Vector(self.Vel.X, self.Vel.Y);
 end
 
 function Update(self)
+	if not self.ToDelete and self.trailPar and MovableMan:IsParticle(self.trailPar) then
+		self.trailPar.Pos = self.Pos - Vector(self.lastVel.X, self.lastVel.Y):SetMagnitude(math.min(self.lastVel.Magnitude * rte.PxTravelledPerFrame, self.trailLength) * 0.5);
+		self.trailPar.Vel = self.lastVel * 0.5;
+		self.trailPar.Lifetime = self.Age + TimerMan.DeltaTimeMS;
+	end
 	if self.deleteNextFrame then
 		self.ToDelete = true;
 	elseif self.Vel.Magnitude >= self.speedThreshold then
 		--Check forward.
 		self.CheckCollision(self.Vel);
```

Comment:
Why not use OnCollideWithMO for this instead of a custom collision check?? Or does it need to do some bigger area than the actual bullet for some reason?

Reply:
Because OnCollideWithMO has been broken for who knows how long. I don't recall touching this part much tho, so it's what cave made initially.

Reply:
Okay, this OnCollideWithMO  problem is probably something that should be fixed if possible.
Maybe it doesn't work well with bullets? I'm curious what you've tested and had work/not work. Let's chat in discord.